app/models/firmware_registry.rb

```diff
@@ -31,6 +31,49 @@ def sync_fw_binaries_raw
     raise NotImplementedError, 'Must be implemented in subclass'
   end
 
+  def self.create_firmware_registry_queue(options)
+    MiqQueue.put_unless_exists(
+      :class_name  => registry_klass(options.delete(:type)).name,
```

Comment:
Hm I don't like that we'd need to update the parent class with knowledge of every child class for this to work.  The more common way to do this would be to have the caller pass in `:type => "FirmwareRegistry::RestApiDepot"` (this works with `.create!`) which you could then pass to `:class_name` there.

Reply:
@agrare fine by me as long as UI people will not be upset to see class name in React where we'll have drop-down to pick registry type (drop-down will display user-friendly version, naturally, but the value will be class name).